Xamarin.Forms.Core/NavigationPage.cs

```diff
@@ -334,7 +335,7 @@ void InsertPageBefore(Page page, Page before)
 			}
 
 			if (PoppedToRoot != null)
-				PoppedToRoot(this, new NavigationEventArgs(root));
+				PoppedToRoot(this, new PoppedToRootEventArgs(root) { PoppedPages = childrenToRemove.OfType<Page>().ToList()});
```

Comment:
I'm wondering about the order of the items in the list. Should it be reversed so that the top most page is first? Does it matter?